hubgrep/models/__init__.py

```diff
@@ -42,8 +47,27 @@ class HostingService(db.Model):
     # api keys for a backend?
     api_url = db.Column(db.String(500), unique=True, nullable=False)
 
+    # individual config for a specific service (eg. api-key)
     # could be json, but thats only supported for postgres
     config = db.Column(db.Text)
 
     # frontend label
     label = db.Column(db.String(80))
+
+    def set_service_label(self):
+        self.label = re.split("//", self.api_url)[1].rstrip("/")
+
+    def get_hosting_service_interface(self, cached_session: 'CachedSession', timeout: int):
```

Comment:
the model object makes the hosting interface now! 
the hosting interface list creation still happens in get_hosting_service_interfaces(), but since hosting interface kind of derives from the model, we do that here.